tests/bot/cogs/test_antimalware.py

```diff
@@ -0,0 +1,146 @@
+        self.message.attachments = [attachment]
+        self.message.channel.send = AsyncMock()
+
+        await self.cog.on_message(self.message)
+        args, kwargs = self.message.channel.send.call_args
+        embed = kwargs.pop("embed")
+
+        self.assertEqual(args[0], f"Hey {self.message.author.mention}!")
+        self.assertEqual(embed.description, (
+            "It looks like you tried to attach a Python file - "
+            f"please use a code-pasting service such as {URLs.site_schema}{URLs.site_paste}"
+        ))
+
+    async def test_txt_file_redirect_embed(self):
+        attachment = MockAttachment(filename="python.txt")
+        self.message.attachments = [attachment]
+        self.message.channel.send = AsyncMock()
+
+        await self.cog.on_message(self.message)
+        args, kwargs = self.message.channel.send.call_args
+        embed = kwargs.pop("embed")
+        cmd_channel = self.bot.get_channel(Channels.bot_commands)
+
+        self.assertEqual(args[0], f"Hey {self.message.author.mention}!")
+        self.assertEqual(embed.description, (
+            "**Uh-oh!** It looks like your message got zapped by our spam filter. "
+            "We currently don't allow `.txt` attachments, so here are some tips to help you travel safely: \n\n"
+            "• If you attempted to send a message longer than 2000 characters, try shortening your message "
+            "to fit within the character limit or use a pasting service (see below) \n\n"
+            "• If you tried to show someone your code, you can use codeblocks \n(run `!code-blocks` in "
+            f"{cmd_channel.mention} for more information) or use a pasting service like: "
+            f"\n\n{URLs.site_schema}{URLs.site_paste}"
+        ))
+
+    async def test_removing_deleted_message_logs(self):
+        """Removing an already deleted message logs the correct message"""
+        attachment = MockAttachment(filename="python.asdfsff")
+        self.message.attachments = [attachment]
+        self.message.delete = AsyncMock(side_effect=NotFound(response=Mock(status=""), message=""))
+
+        logger = logging.getLogger(MODULE)
+
+        with self.assertLogs(logger=logger, level="INFO") as logs:
+            await self.cog.on_message(self.message)
+        self.assertIn(
+            f"INFO:{MODULE}:Tried to delete message `{self.message.id}`, but message could not be found.",
+            logs.output)
+
+    async def test_message_with_illegal_attachment_logs(self):
+        """Deleting a message with an illegal attachment should result in a log."""
+        attachment = MockAttachment(filename="python.asdfsff")
+        self.message.attachments = [attachment]
+
+        file_extensions = {splitext(attachment.filename.lower())[1] for attachment in self.message.attachments}
+        extensions_blocked = file_extensions - set(AntiMalwareConfig.whitelist)
+        blocked_extensions_str = ', '.join(extensions_blocked)
```

Comment:
Though I don't think the log's contents should be assert at all, I'll mention this anyway:

You shouldn't copy the same code into tests. If the cog had this code in a function, you'd be able to mock the function's return value and then assert your given return value was within the log string.